src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -429,7 +429,7 @@ public static SyntaxToken WithoutTrivia(this SyntaxToken token)
         /// </summary>
         internal static SyntaxNode AsRootOfNewTreeWithOptionsFrom(this SyntaxNode node, SyntaxTree oldTree)
         {
-            return oldTree.WithRootAndOptions(node, oldTree.Options).GetRoot();
+            return node != null ? oldTree.WithRootAndOptions(node, oldTree.Options).GetRoot() : null;
```

Comment:
Why do we need the null check now but not befores;

Reply:
This check prevents a NRE when the removing the root node, see [this comment](https://github.com/dotnet/roslyn/pull/23165#discussion_r190006149) for more details.